language/vm/vm-runtime/src/data_cache.rs

```diff
@@ -90,6 +90,7 @@ pub struct TransactionDataCache<'txn> {
     // case moving forward, so we need to review this.
     // Also need to relate this to a ResourceKey.
     data_map: BTreeMap<AccessPath, GlobalRef>,
+    module_map: BTreeMap<ModuleId, Vec<u8>>,
```

Comment:
adding a cache for modules make producing the `WriteSet` more "canonical", that is, without having to carry the modules around which felt odd.
Of course there will only be one module in this cache, given the semantic of our transactions but I think this is good enough and also something genesis generation is taking advantage of

Reply:
Btw can you confirm if the genesis will remain to be unchanged if you just keep the insertion order just to make sure the refactor is not messing other parts up?